hashicorp/boundary-ui
# fix: 🐛 appUpdater support ARM64

# Description
Add ARM64 support for the appUpdater

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-13666)


## How to Test

### Test 1:
- Within dev environment, start the app.
- When the autoUpdater launches, say YES (so it downloads the latest version.
- Make sure on the terminal the donwloaded files is the latet published DC
![Screenshot 2024-06-10 at 2 45 04 PM](https://github.com/hashicorp/boundary-ui/assets/9775006/5344d5e4-b4c9-404d-8127-17cbf87ace85)

### Test 2: (to be run ONLY on ARM chips)
- Grab ARM64 binary [from this test build](https://github.com/hashicorp/boundary-ui-releases/actions/runs/9667402379).
- Install the test binary within your system (drag & drop to apps folder).
- Run the test binary.
- The test build MUST trigger the autoupdater, since the version set on the test build `1.2.3` is lower than the latest publish version `2.0.3`.
- Say YES to update.
- Check the version has been updated to ARM64 DC `2.0.3`.

### Test 3: (to be run ONLY on ARM chips)
This test checks a very important step, if the user has the AMD64 but its arch is ARM64, we must swap from AMD to ARM

- Grab AMD64 (x64) binary [from this test build](https://github.com/hashicorp/boundary-ui-releases/actions/runs/9667402379).
- Install the test binary within your system (drag & drop to apps folder).
- Run the test binary.
- The test build MUST trigger the autoupdater, since the version set on the test build 1.2.3 is lower than the latest publish version 2.0.3.
- Say YES to update.
- Check the version has been updated to ARM64 DC 2.0.3.

### Test 4: (to be run ONLY on INTEL chips)
- Grab AMD64 (x64) binary [from this test build](https://github.com/hashicorp/boundary-ui-releases/actions/runs/9667402379).
- Install the test binary within your system (drag & drop to apps folder).
- Run the test binary.
- The test build MUST trigger the autoupdater, since the version set on the test build `1.2.3` is lower than the latest publish version `2.0.3`.
- Say YES to update.
- Check the version has been updated to AMD64 DC `2.0.3`.
